cudapoa/src/main.cpp

```diff
@@ -0,0 +1,276 @@
+                                                batch_size,
+                                                gap_score,
+                                                mismatch_score,
+                                                match_score,
+                                                banded_alignment);
+
+    return std::move(batch);
+}
+
+void process_batch(Batch* batch, bool msa, bool print)
+{
+    batch->generate_poa();
+
+    StatusType status = StatusType::success;
+    if (msa)
+    {
+        // Grab MSA results for all POA groups in batch.
+        std::vector<std::vector<std::string>> msa; // MSA per group
+        std::vector<StatusType> output_status;     // Status of MSA generation per group
+
+        status = batch->get_msa(msa, output_status);
+        if (status != StatusType::success)
+        {
+            std::cerr << "Could not generate MSA for batch : " << status << std::endl;
+        }
+
+        for (int32_t g = 0; g < get_size(msa); g++)
+        {
+            if (output_status[g] != StatusType::success)
+            {
+                std::cerr << "Error generating  MSA for POA group " << g << ". Error type " << output_status[g] << std::endl;
+            }
+            else
+            {
+                if (print)
+                {
+                    for (const auto& alignment : msa[g])
+                    {
+                        std::cout << alignment << std::endl;
+                    }
+                }
+            }
+        }
+    }
+    else
+    {
+        // Grab consensus results for all POA groups in batch.
+        std::vector<std::string> consensus;          // Consensus string for each POA group
+        std::vector<std::vector<uint16_t>> coverage; // Per base coverage for each consensus
+        std::vector<StatusType> output_status;       // Status of consensus generation per group
+
+        status = batch->get_consensus(consensus, coverage, output_status);
+        if (status != StatusType::success)
+        {
+            std::cerr << "Could not generate consensus for batch : " << status << std::endl;
+        }
+
+        for (int32_t g = 0; g < get_size(consensus); g++)
+        {
+            if (output_status[g] != StatusType::success)
+            {
+                std::cerr << "Error generating consensus for POA group " << g << ". Error type " << output_status[g] << std::endl;
+            }
+            else
+            {
+                if (print)
+                {
+                    std::cout << consensus[g] << std::endl;
+                }
+            }
+        }
+    }
+}
+
+int main(int argc, char* argv[])
+{
+    // Parse input parameters
+    const ApplicationParameters parameters(argc, argv);
+
+    // Load input data. Each window is represented as a vector of strings. The sample
+    // data has many such windows to process, hence the data is loaded into a vector
+    // of vector of strings.
+    std::vector<std::vector<std::string>> windows;
```

Comment:
as discussed earlier, let's rename `windows` to `groups`

Reply:
just to clarify a bit: from the changed files view it's not obvious, but in the `main.cpp`, file parser stores data in a vector of `strings` named `windows`, after that converts them into an array of `groups`. Regarding the interface, I see they are already changed to `groups`, e.g. [https://github.com/atadkase/ClaraGenomicsAnalysis/blob/atadkase/cudapoa-bin/cudapoa/src/application_parameters.cpp#L196](url)

Reply:
Yeah. I renamed the external interfaces to groups. Didn't percolate it through the rest of the code though.

Reply:
Okay perfect, thanks!